### PR TITLE
fix reference links in lessons.html

### PR DESCRIPTION
**Pull request by tjhei**
The current links give 404s for me.

### CODE REVIEW

File: pages/lessons.html [hunk 1]
```diff
@@ -49,7 +49,7 @@ <h4>Contributing</h4>
     <td id="shell">The Unix Shell</td>
     <td><a href="{{site.github_io_url}}/shell-novice" target="_blank" class="icon-browser" title="The Unix Shell"></a></td>
     <td><a href="{{site.github_url}}/shell-novice" target="_blank" class="icon-github" title="The Unix Shell"></a></td>
-    <td><a href="{{site.github_io_url}}/shell-novice/reference/" target="_blank" class="icon-eye" title="The Unix Shell"></a></td>
+    <td><a href="{{site.github_io_url}}/shell-novice/reference" target="_blank" class="icon-eye" title="The Unix Shell"></a></td>
```
Comment (rgaiacs) [inline pages/lessons.html:52]: Need to be `{{site.github_io_url}}/shell-novice/reference.html`.

File: pages/lessons.html [hunk 2]
```diff
@@ -135,7 +135,7 @@ <h4>Contributing</h4>
     <td>Instructor Training</td>
     <td><a href="https://carpentries.github.io/instructor-training/"_blank" class="icon-browser" title="Instructor Training"></a></td>
     <td><a href="https://github.com/carpentries/instructor-training" target="_blank" class="icon-github" title="Instructor Training"></a></td>
-    <td><a href="https://carpentries.github.io/instructor-training/reference/" target="_blank" class="icon-eye" title="Instructor Training"></a></td>
+    <td><a href="https://carpentries.github.io/instructor-training/reference" target="_blank" class="icon-eye" title="Instructor Training"></a></td>
```
Comment (rgaiacs) [inline pages/lessons.html:138]: Need to be `https://carpentries.github.io/instructor-training/reference.html`

File: pages/lessons.html [hunk 3]
```diff
@@ -126,7 +126,7 @@ <h4>Contributing</h4>
     <td>Automation and Make</td>
     <td><a href="{{site.github_io_url}}/make-novice" target="_blank" class="icon-browser" title="Automation and Make"></a></td>
     <td><a href="{{site.github_url}}/make-novice" target="_blank" class="icon-github" title="Automation and Make"></a></td>
-    <td><a href="{{site.github_io_url}}/make-novice/reference/" target="_blank" class="icon-eye" title="Automation and Make"></a></td>
+    <td><a href="{{site.github_io_url}}/make-novice/reference" target="_blank" class="icon-eye" title="Automation and Make"></a></td>
```
Comment (rgaiacs) [inline pages/lessons.html:129]: Need to be `{{site.github_io_url}}/make-novice/reference.html`

**Review by rgaiacs — CHANGES_REQUESTED** [review]
This is related with the last change in the template/styles. Other lessons also need a similar change.

**Comment by tjhei** [comment]
> Need to be {{site.github_io_url}}/shell-novice/reference.html.

My proposed change to https://swcarpentry.github.io/shell-novice/reference does work though (without the .html). I can add the ``.html`` though if you prefer.

> Other lessons also need a similar change.

For me only those three links in that table needed fixing.

**Comment by rgaiacs** [comment]
@tjhei Can you remove the trailing slashes from the other links? Based on Stackoverflow, see https://stackoverflow.com/questions/33270605/github-pages-trailing-slashes, the web server should be smart enough to sort out the correct file to return without the trailing slash but looks like that, as you discover, the trailing slash will enforce the server to only look for the `index.html` file inside the directory `reference` which doesn't exist, which is a good behaviour by the way.

**Comment by tjhei** [comment]
> Can you remove the trailing slashes from the other links?

I can do that. They get forwarded to ``.../reference/`` then anyways. Note that the links in the top menu under "extras" link to ``reference.html`` for the three entries I fixed and to ``reference/`` (with a trailing slash) for the others. Not sure if that changes your mind on this. ;-)

**Comment by rgaiacs** [comment]
@tjhei For clarification! :-)

All link in https://software-carpentry.org/lessons/, the file that you are editing, should have the reference link to `reference` with**out** the trailing slash. It should work for lessons that didn't merge the new style and also for the one that already merged the new style.

> Note that the links in the top menu under "extras"

See https://github.com/swcarpentry/styles/blob/gh-pages/_includes/navbar.html#L61. We are using the "new" `link` function so the link will be correct.

**Comment by tjhei** [comment]
done

**Pull request merged**